src/components/Details/VmDetails/VmDetails.js

```diff
@@ -0,0 +1,252 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { forEach, get, has } from 'lodash';
+import { FieldLevelHelp } from 'patternfly-react';
+import { PodModel, VirtualMachineInstanceMigrationModel, VirtualMachineModel } from '../../../models';
+import { VmStatus } from '../../VmStatus';
+
+const DASHES = '---';
+const VIRT_LAUNCHER_POD_PREFIX = 'virt-launcher-';
+const IMPORTER_DV_POD_PREFIX = 'importer-datavolume-';
+
+const cloudInitInUse = vm => {
+  let inUse = false;
+
+  // Get volumes with 'cloudInitNoCloud' property
+  const volumes = get(vm, 'spec.template.spec.volumes', []);
+  const cloudInitVolumes = volumes
+    .map(vol => (has(vol, 'cloudInitNoCloud') ? vol.name : null))
+    .filter(name => name !== null);
+
+  if (cloudInitVolumes.length === 0) {
+    return inUse;
+  }
+
+  // Check for disks using a volume with 'cloudInitNoCloud' property
+  const disks = get(vm, 'spec.template.spec.domain.devices.disks', []);
+  inUse = forEach(disks, disk => cloudInitVolumes.includes(disk.volumeName)).reduce(
+    (acc, currValue) => acc && currValue
+  );
+
+  return inUse;
+};
+
+const getPod = (vm, resources, podNamePrefix) => {
+  const podData = getFlattenForKind(PodModel.kind, resources);
+  return findPod(podData, vm.metadata.name, podNamePrefix);
+};
+
+const getMigration = (vm, resources) => {
+  const migrationData = getFlattenForKind(VirtualMachineInstanceMigrationModel.kind, resources);
+  return findVMIMigration(migrationData, vm.metadata.name);
+};
+
+export const findVMIMigration = (data, vmiName) => {
+  if (!data || !data.items) {
+    return null;
+  }
+  const migrations = data.items.filter(m => m.spec.vmiName === vmiName);
+  return migrations.find(m => !get(m, 'status.completed') && !get(m, 'status.failed'));
+};
+
+const StateColumn = props => {
```

Comment:
You could use VmStatus from https://github.com/kubevirt/web-ui-components/blob/master/src/components/VmStatus/VmStatus.js

Reply:
Done!